### PR TITLE
Call cargo directly instead of using actions-rs/cargo

### PR DESCRIPTION
**Pull request by lnicola**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [ ] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

`actions-rs` is unmaintained, and doesn't bring us that much. I think we might lose the nice GitHub integration, but it wasn't really working anyway.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,25 +41,14 @@ jobs:
...
-        with:
-          command: fmt
-          args: --all -- --check
+        run: cargo fmt --all --check
       - name: Check with Clippy
-        uses: actions-rs/clippy-check@v1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: --all-targets ${{ matrix.features }} -- -D warnings
+        run: cargo clippy
```
Comment (jdroenner) [inline .github/workflows/ci.yml:46]: I think clippy should get "--all-targets" and "${{ matrix.features }}" otherwise it might skip tests and other code.

Comment (lnicola) [reply .github/workflows/ci.yml:46]: Oops, not sure how I lost that :grimacing:.

Comment (lnicola) [reply .github/workflows/ci.yml]: Fixed, but now CI fails because we have a ton of clippy warnings.

Comment (lnicola) [reply .github/workflows/ci.yml]: Filed #326. It doesn't actually fail (yet), I was using `beta`.

**Review by jdroenner — APPROVED** [review]

**Comment by jdroenner** [comment]
bors merge

**Comment by lnicola** [comment]
bors r=jdroenner

**Comment by lnicola** [comment]
bors retry

**Comment by lnicola** [comment]
bors r=jdroenner

**Comment by lnicola** [comment]
bors retry

**Pull request merged**